src/api-service/__app__/onefuzzlib/orm.py

```diff
@@ -66,6 +67,26 @@
 HOURS = 60 * 60
 
 
+class HasState(Protocol):
+    # TODO: this should be bound tighter than Any
+    state: Any
+
+
+def process_update(obj: HasState) -> None:
+    """ process through the state machine for an object """
+
+    # process the state machine up to 5 times unless it's stopped changing
+    for _ in range(0, 5):
```

Comment:
1. Why this choice of 5? 

2. The name is singular, but this might process many updates, right? It would be nice to have separate methods for (1) a single step of updating, returning a bool if the state changed, and (2) a wrapper, equivalen to this funciton, that calls the singel-step function in a loop, with e.g. a `max_updates` kwarg that defaults to 5.

3. More generally, does it may sense to just have this be part of one of the ORM classes?